x/ibc/07-tendermint/misbehaviour.go

```diff
@@ -0,0 +1,90 @@
+package tendermint
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	clientexported "github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
+	ibctypes "github.com/cosmos/cosmos-sdk/x/ibc/types"
+)
+
+// CheckMisbehaviourAndUpdateState determines whether or not two conflicting
+// headers at the same height would have convinced the light client.
+//
+// NOTE: assumes provided height is the height at which the consensusState is
+// stored.
+func CheckMisbehaviourAndUpdateState(
+	clientState clientexported.ClientState,
+	consensusState clientexported.ConsensusState,
+	misbehaviour clientexported.Misbehaviour,
+	height uint64, // height at which the consensus state was loaded
```

Comment:
OK, is that guaranteed somewhere else?

Reply:
on the caller function within ICS02

Reply:
Hmm ok. This pattern strikes me as easy to get wrong (since the checks are split)

Reply:
Doesn't look like it is:
https://github.com/cosmos/cosmos-sdk/pull/5485/files#diff-be64020aa203f351d8bd639c630aa097R124

Since the height passed in is the height of the misbehaviour not the height of the consensusState. While this PR makes them the same height, i believe they may not be in general

Reply:
>Since the height passed in is the height of the misbehaviour not the height of the consensusState.

@AdityaSripal the consensus state is loaded at the misbehaviour height and the tx will fail if no consensus state is found. 

> While this PR makes them the same height, i believe they may not be in general

Should I change the height on the caller to smth else?